clang-tools-extra/test/clang-doc/json/class-requires.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: clang-doc --extra-arg -std=c++20 --output=%t --format=json --executor=standalone %s
+// RUN: FileCheck %s < %t/GlobalNamespace/MyClass.json
+
+template<typename T>
+concept Addable = requires(T a, T b) {
+  { a + b };
+};
+
+template<typename T>
+requires Addable<T>
+struct MyClass {
+};
+
+// CHECK:       "Name": "MyClass",
+// CHECK-NEXT:  "Namespace": [
+// CHECK-NEXT:    "GlobalNamespace"
+// CHECK-NEXT:  ],
+// CHECK-NEXT:  "Path": "GlobalNamespace",
+// CHECK-NEXT:  "TagType": "struct",
+// CHECK-NEXT:  "Template": {
+// CHECK-NOT:     "Constraints": [
+// CHECK-NOT:       "Addable<T>"
```

Comment:
Interesting thing about Constraints in the context of Clang-Doc is that we don't get back any Decls AFAIK, so I can't construct this as a Reference like BaseRecords. Which is unfortunate in terms of linking. Might be able to find a way to do this but I don't know much about how `Expr` works.

Reply:
Thanks for pointing this out. As discussed offline, let’s ask clang maintainers on discord or discourse about how to handle this. For this patch things are fine though since it’s just documenting the current behavior.

Reply:
I changed the tests so that the constraints are now arrays of Infos. Implementation is up the stack.